File: rotate.c
void rotate(int* nums, int numsSize, int k) {
    if(!(k %= numsSize)) return;
    int index = 0, index2 = numsSize-k, n = numsSize - 1;
    while(n-- > 0){
        if( index == index2 ){
            index = (++index2) + k - numsSize;
            n--;
        }
        nums[index] ^= nums[index2];
        nums[index2] ^= nums[index];
        nums[index] ^= nums[index2];
        index = (index + k) % numsSize;
    }
}

File: rotate.js
var rotate = function(matrix) {
    matrix.reverse()
    for (let i = 0; i < matrix.length; i++) {
        for (let j = i+1; j < matrix[i].length; j++) {
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]]
        }
    }
    return matrix
};